pkg/querier/tenantfederation/tenant_federation.go

```diff
@@ -7,8 +7,11 @@ import (
 type Config struct {
 	// Enabled switches on support for multi tenant query federation
 	Enabled bool `yaml:"enabled"`
+	// MaxConcurrent The number of workers used for processing federated query.
+	MaxConcurrent int `yaml:"max_concurrent"`
 }
 
 func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.BoolVar(&cfg.Enabled, "tenant-federation.enabled", false, "If enabled on all Cortex services, queries can be federated across multiple tenants. The tenant IDs involved need to be specified separated by a `|` character in the `X-Scope-OrgID` header (experimental).")
+	f.IntVar(&cfg.MaxConcurrent, "tenant-federation.max-concurrent", defaultMaxConcurrency, "The number of workers used for processing federated query.")
```

Comment:
This is the number of workers for a single federated query, correct? The description is a bit misleading as it sounds like concurrency for shared for all federated queries

Reply:
what about `The number of workers used to process each federated query.`?